sklearn/elasticnet.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (C) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import argparse
+from bench import (
+    parse_args, measure_function_time, load_data, print_output, rmse_score
+)
+from sklearn.linear_model import ElasticNet
+
+parser = argparse.ArgumentParser(description='scikit-learn elastic-net regression '
+                                             'benchmark')
+parser.add_argument('--no-fit-intercept', dest='fit_intercept', default=False,
+                    action='store_false',
+                    help="Don't fit intercept (assume data already centered)")
+parser.add_argument('--alpha', dest='alpha', type=float, default=1.0,
+                    help='Regularization parameter')
+parser.add_argument('--maxiter', type=int, default=1000,
+                    help='Maximum iterations for the iterative solver')
+parser.add_argument('--l1_ratio', dest='l1_ratio', type=float, default=0.5,
+                    help='Regularization parameter')
+parser.add_argument('--tol', type=float, default=0.0,
+                    help='Tolerance for solver.')
+params = parse_args(parser)
+
+# Load data
+X_train, X_test, y_train, y_test = load_data(params)
+
+# Create our regression object
+regr = ElasticNet(fit_intercept=params.fit_intercept, l1_ratio=params.l1_ratio, alpha=params.alpha,
+                        tol=params.tol, max_iter=params.maxiter, copy_X=False)
+
+columns = ('batch', 'arch', 'prefix', 'function', 'threads', 'dtype', 'size',
+           'time')
+
+# Time fit
+fit_time, _ = measure_function_time(regr.fit, X_train, y_train, params=params)
+
+# Time predict
+predict_time, pred_train = measure_function_time(regr.predict, X_train, params=params)

```

Comment:
Are you sure that we should measure prediction with training dataset? As I see, in other algorithms we use testing dataset for it. (The same question for Lasso and SVM)

Reply:
I would like to measure predictions everywhere on the training set. this is more data, but take into account the test dataset to compare the accuracy. But old measurements and benchmarks cannot be updated - since we are already targeting these cases. But algorithms Lasso, Elastic-Net, SVM - new.